testing/servers_test.go

```diff
@@ -0,0 +1,152 @@
+package testing
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/onsi/gomega/ghttp"
+	v1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+
+	. "github.com/onsi/ginkgo/v2/dsl/core" // nolint
+	. "github.com/onsi/gomega"             // nolint
+)
+
+var _ = Describe("RespondWithOCMObjectMarshal", func() {
+	var server *ghttp.Server
+
+	BeforeEach(func() {
+		server = MakeTCPServer()
+	})
+
+	AfterEach(func() {
+		server.Close()
+	})
+
+	It("should respond with JSON encoded object using provided marshal function", func() {
+		// Create a mock cluster object (we'll use a simple struct for testing)
+		cluster, err := v1.NewCluster().
+			ID("test-cluster").
+			Name("Test Cluster").
+			Build()
+		Expect(err).ToNot(HaveOccurred())
+
+		// Configure the server to respond with the cluster
+		server.AppendHandlers(
+			RespondWithOCMObjectMarshal(200, cluster, v1.MarshalCluster),
+		)
+
+		// Make a request
+		resp, err := http.Get(server.URL())
+		Expect(err).ToNot(HaveOccurred())
+		defer resp.Body.Close()
+
+		// Verify response
+		Expect(resp.StatusCode).To(Equal(200))
+		Expect(resp.Header.Get("Content-Type")).To(Equal("application/json"))
+
+		// Read and verify the response body contains JSON
+		body, err := io.ReadAll(resp.Body)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(string(body)).To(ContainSubstring("test-cluster"))
+		Expect(string(body)).To(ContainSubstring("Test Cluster"))
+	})
+
+	It("should handle nil objects gracefully", func() {
+		server.AppendHandlers(
+			RespondWithOCMObjectMarshal(200, nil, v1.MarshalCluster),
+		)
+
+		resp, err := http.Get(server.URL())
+		Expect(err).ToNot(HaveOccurred())
+		defer resp.Body.Close()
+
+		Expect(resp.StatusCode).To(Equal(500))
+	})
+
+	It("should handle invalid marshal function", func() {
+		cluster := &v1.Cluster{}
+		server.AppendHandlers(
+			RespondWithOCMObjectMarshal(200, cluster, "not-a-function"),
+		)
+
+		resp, err := http.Get(server.URL())
+		Expect(err).ToNot(HaveOccurred())
+		defer resp.Body.Close()
+
+		Expect(resp.StatusCode).To(Equal(500))
+	})
+})
+
+// TestRespondWithOCMObjectMarshal provides unit tests for the marshal function
+func TestRespondWithOCMObjectMarshal(t *testing.T) {
+	// Create a test cluster
+	cluster, err := v1.NewCluster().
+		ID("test-cluster-123").
+		Name("Test Cluster for Unit Test").
+		Build()
+	if err != nil {
+		t.Fatalf("Failed to create test cluster: %v", err)
+	}
+
+	// Create a test request
+	req := httptest.NewRequest("GET", "/test", nil)
+	rr := httptest.NewRecorder()
```

Comment:
Would like to avoid single or 2 char variable - those are hard to read and follow.